ConsentViewController/Classes/SourcePointClient/Adapters.swift

```diff
@@ -165,12 +165,82 @@ extension ConsentStatus.GranularStatus {
     }
 }
 
+extension SPGPPConfig.SPMspaBinaryFlag {
+    func toCore() -> SPMobileCore.IncludeData.MspaBinaryFlag? {
+        switch self {
+        case .yes: return .yes
+        case .no: return .no
+        }
+    }
+}
+
+extension SPGPPConfig.SPMspaTernaryFlag {
+    func toCore() -> SPMobileCore.IncludeData.MspaTernaryFlag? {
+        switch self {
+        case .yes: return .yes
+        case .no: return .no
+        case .notApplicable: return .na
+        }
+    }
+}
+
+extension IncludeData {
+    func toCore() -> SPMobileCore.IncludeData {
+        var translateMessageVal = nil as Bool?
+        #if os(tvOS)
+        translateMessageVal = translateMessage
+        #endif
+        return SPMobileCore.IncludeData.init(
+            tcData: SPMobileCore.IncludeData.TypeString(type: "RecordString"),
+            webConsentPayload: SPMobileCore.IncludeData.TypeString(type: "string"),
+            localState: SPMobileCore.IncludeData.TypeString(type: "RecordString"),
+            categories: categories,
+            translateMessage: KotlinBoolean(bool: translateMessageVal),
+            gppData: SPMobileCore.IncludeData.GPPConfig(
+                MspaCoveredTransaction: gppConfig.MspaCoveredTransaction?.toCore(),
+                MspaOptOutOptionMode: gppConfig.MspaOptOutOptionMode?.toCore(),
+                MspaServiceProviderMode: gppConfig.MspaServiceProviderMode?.toCore(),
+                uspString: KotlinBoolean(bool: gppConfig.uspString)
+            )
+        )
+    }
+}
+
```

Comment:
@Nevazhnovu maybe this can be deleted?

Reply:
@andresilveirah I tried to delete it, but it doesn't work since in `0.0.10`, `postChoice`s demand `IncludeData` to be added to the Request object in initialization.